test/plugins/test-trace-knex.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var common = require('./common.js');
+var traceLabels = require('../../src/trace-labels.js');
+
+var RESULT_SIZE = 5;
+var TABLE_NAME = 't';
+
+var obj = {
+  k: 1,
+  v: 'obj'
+};
+
+var versions = {
+  knex10: './fixtures/knex0.10',
+  knex11: './fixtures/knex0.11',
+  knex12: './fixtures/knex0.12',
+  knex13: './fixtures/knex0.13'
+};
+
+describe('test-trace-knex', function() {
+  var agent;
+  var assert;
+  before(function() {
+    agent = require('../..').start({
+      logLevel: 2,
+      flushDelaySeconds: 1,
+      enhancedDatabaseReporting: true,
+      databaseResultReportingSize: RESULT_SIZE
+    });
+    assert = require('assert');
```

Comment:
Can this be moved to the top level. Assert does not hold any state that could interfere with the trace agent's patching.